src/test/java/com/Model/ControllerTest/AdminTest.java

```diff
@@ -66,31 +72,65 @@ public void AdminLoginTest () throws Exception {
 
     }
 
+    /**
+     * Test admin log in with wrong credentials should not be authenticated
+     * @throws Exception
+     */
+    @Test
+    public void AdminLoginTestWithWrongCredentials () throws Exception {
+        AdminUser adminUser = new AdminUser("test@test.test", "testpassword");
+        this.userRepository.save(adminUser);
+        RequestBuilder requestBuilder = formLogin().user("emailaddress","test@test.test").password("password","testpassword222").loginProcessingUrl("/userlogin");
+        this.mockMvc.perform(requestBuilder)
+                .andExpect(unauthenticated());
+
+    }
+
+    /**
+     * User with "ADMIN_Role" should get access to admin page
+     * @throws Exception
+     */
     @Test
     @WithMockUser(username="admin",roles={"ADMIN"})
     public void adminPageTest () throws Exception {
         this.mockMvc.perform(get("/admin"))
                 .andExpect(status().isOk());
     }
 
+    /**
+     * Normal user should not get access to admin page
+     * @throws Exception
+     */
     @Test
     @WithMockUser(username="user",roles={"USER"})
     public void adminPageTestWithoutAdmin () throws Exception {
         this.mockMvc.perform(get("/admin"))
                 .andExpect(status().isForbidden());
     }
 
+    /**
+     * User with Admin_Role should be able to approve applicants
+     * Once applicants approved, it should exist in Userrepository.
+     * @throws Exception
+     */
     @Test
     @WithMockUser(username = "admin", roles = {"ADMIN"})
     public void adminAppprove () throws Exception {
         Applicant applicant = this.applicantRepository.save(new Applicant(this.emailAddress, "description text"));
         long id = applicant.getId();
+        assertEquals (false, this.userRepository.findUserByEmailAddressIgnoreCase(this.emailAddress)!=null);
         String data = "\"" + String.valueOf(id) + "\"";
         this.mockMvc.perform(post("/admin/approveapplicant").content(data).contentType("application/json"))
                 .andExpect(status().isOk());
         assertEquals (false, this.applicantRepository.existsById(id));
+        assertEquals (true, this.userRepository.findUserByEmailAddressIgnoreCase(this.emailAddress)!=null);
```

Comment:
what can the applicant do once he/she is approved?

Reply:
The applicant is able to login with a provided credential and creates verified events, which is tested in UserLoginTest and EventTest.